### ISSUE TITLE
String compression

### ISSUE THREAD

**Issue by JohannesLichtenberger** [sirixdb/sirix#556]
[FSST: Fast Random Access String Compression](https://www.vldb.org/pvldb/vol13/p2649-boncz.pdf)

**Comment by JohannesLichtenberger** [comment]
Useable either as a replacement of the current dictionary encoding (used to store object field names or element names/attribute names) or simply for the text compression of string values.

**Comment by AlvinKuruvilla** [comment]
I feel like the best way to implement this is to keep this as a separate library that we can then as add a dependency. I also found the original c++ code used in the [paper](https://github.com/cwida/fsst)

**Comment by AlvinKuruvilla** [comment]
This is the [current](https://github.com/AlvinKuruvilla/frasc) implementation I have... if we want, we can migrate this to the Sirix organization and continue development from

**Comment by JohannesLichtenberger** [comment]
I think you could simply develop it as a separate library and I can add it as a dependency once it's finished and published to maven central :-)

**Comment by AlvinKuruvilla** [comment]
> I think you could simply develop it as a separate library and I can add it as a dependency once it's finished and published to maven central :-)

Good point

**Comment by JohannesLichtenberger** [comment]
@AlvinKuruvilla did you made any advances?

**Comment by JohannesLichtenberger** [comment]
@AlvinKuruvilla *ping* :-)

**Comment by AlvinKuruvilla** [comment]
Sorry @JohannesLichtenberger, not lately. I just finished school. I hope to get some more work done now that I have some free time. I have a feeling this is going to be a longer-term issue, especially with all of the tests that need to be ported over

**Comment by Aminmalek** [comment]
is this issue fixed?
@JohannesLichtenberger

**Comment by JohannesLichtenberger** [comment]
No, still open. But I think low priority

**Comment by AlvinKuruvilla** [comment]
Yeah, sorry about that @JohannesLichtenberger , I didn't mean to ghost you like that. I've been busy with school and those projects. I'm still interested in the project just haven't had the time as of late., I have most of the basic building blocks coded up for the most part, and I had some tests written up. Can we consider making this part of the org? It's nowhere near ready or stable, but I'm familiar enough with the codebase to set things up so people can at least look at open issues and consider contributing. That way, we can drive some progress when I can't work on it.

**Comment by JohannesLichtenberger** [comment]
@Aminmalek wanted to work on this, so you may transfer it to the organization.

**Comment by Aminmalek** [comment]
@AlvinKuruvilla we can work on this together.
